src/Build/Evaluation/Expander.cs

```diff
@@ -1125,7 +1125,7 @@ private static class PropertyExpander<T>
                     // (but not including) the "$(", and advance the sourceIndex pointer.
                     if (propertyStartIndex - sourceIndex > 0)
                     {
-                        results.Add(expression.Substring(sourceIndex, propertyStartIndex - sourceIndex));
+                        results.Add(expression.AsMemory(sourceIndex, propertyStartIndex - sourceIndex));
```

Comment:
Does this need a CLR2 compatibility check?

Reply:
It does not because `MSBuildTaskHost is not doing evaluation and this file is not compiled for CLR2.